### PR TITLE
Add example of horizontal orientation to violin

### PR DESCRIPTION
**Pull request by kescobo**
I always get confused about which arguments are switched (the values passed, here `xs` and `ys`) and which arguments stay the same (eg `ylable`  still applies to the vertical axis). 

Let me know if this is useful, and I'm happy to add a similar thing to `boxplot`.

### CODE REVIEW

File: docs/reference/plots/violin.md [hunk 1]
```diff
@@ -94,3 +94,33 @@ violin(fig[1,2], x, y, weights = w)
...
+#### Horizontal axis
+
+\begin{examplefigure}{}
+```julia
+using CairoMakie
+CairoMakie.activate!() # hide
+
+fig = Figure()
+
+xs = rand(1:3, 1000)
```
[the remark this answers is not in the document]
Comment (kescobo) [reply docs/reference/plots/violin.md:107]: As part of this PR, or separate?

[the remark this answers is not in the document]
Comment (jkrumbiegel) [reply docs/reference/plots/violin.md:107]: You could simply edit this one further

**Comment by kescobo** [comment]
OK, I went ahead and updated `boxplot`, and part of `barplot` as well. There's already an example of `direction=:x` for the later; separately, is there a reason the language here isn't consistent (eg why not `orientation=:horizontal` as in `violin` (or vice versa))?

**Comment by jkrumbiegel** [comment]
> is there a reason the language here isn't consistent

Just that these were written independently, at different points in time. If we can come up with a good way to name all of these, I'd be all for it. We could probably go a deprecation route then and only use the new keyword if the old one isn't `automatic`, we've done that in earlier cases, too.

I was actually wondering if a simple `flip = true` keyword would suffice. This would mean each recipe has a "canonical" orientation, and `flip` just uses the other one. Wouldn't work for 3d, but I don't think there's any recipes to which that would apply.

**Comment by kescobo** [comment]
>I was actually wondering if a simple flip = true keyword would suffice

`flip` seems a bit ambiguous to me, but I could be into it. I remember a lot of bikeshedding around that for Plots.jl, and in any case it's probably orthogonal to this PR. I'll open an issue to track it.

**Comment by kescobo** [comment]
Actually, there's already https://github.com/MakieOrg/Makie.jl/issues/926

**Comment by ffreyer** [comment]
Should we merge this?

**Review by jkrumbiegel — APPROVED** [review]

**Comment by jkrumbiegel** [comment]
Thanks @kescobo!

**Pull request merged**